### ISSUE TITLE
Enum iteration

### ISSUE THREAD

**Issue by nevf** [phenax/enum-fp#19]
Enum-fp looks interesting, but I have a simple question which I've not been able to resolve so far.

Given: `const Action = Enum([ 'Add', 'Edit', 'Delete', 'Get' ]);`

How do I iterate over` Action` to get each enum key?

**Comment by phenax** [comment]
There is no way to do that currently. But it can be easily implemented by adding typeNames to the return instance.

https://github.com/phenax/enum-fp/blob/master/src/utils.js#L37

If you are interested in submitting a PR for this then let me know.

**Comment by nevf** [comment]
@phenax Thanks for the prompt reply. I'm a bit out of my depth with FP and understanding `enum-fp`. Could you possibly add this. Thx.

**Comment by phenax** [comment]
Sure @nevf. I'll release a patch soon.

**Comment by nevf** [comment]
@phenax Great, look forward to checking it out.

**Comment by phenax** [comment]
@nevf Can you tell me the exact use case this is required for? Its so that I can expose a better api.

**Comment by phenax** [comment]
Let me know if #20 fits the use case

**Comment by nevf** [comment]
@phenax Yes thanks, that seems fine. It took me a while to get it working as the `build` code isn't updated.
What would be really nice is to be able to use `forEach`:
````
const Action = Enum([ 'Add', 'Edit', 'Delete', 'Get' ]);
Action.forEach( enum => { ... } )
````

**Comment by phenax** [comment]
Can you clarify the exact use case for me? forEach operation doesn't seem right for a sum type.

**Comment by nevf** [comment]
@phenax I'm using `enum-fp` for its enum capabilities. I'm looking at replacing https://github.com/adrai/enum and/or https://github.com/rauschma/enumify with something with less code and simpler and thought `enum-fp` might fit the bill. I have really looked at the `sum-type` capability which could also be useful. HTH.

**Comment by phenax** [comment]
This has been shipped with the release 1.0.2
